src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1140,17 +1140,23 @@ public ResultSet getProcedureColumns(
       String procedureNameUnparsed = resultSetStepOne.getString("arguments").trim();
       String procedureNameNoArgs = resultSetStepOne.getString("name");
       String schemaName = resultSetStepOne.getString("schema_name");
+      // Check that schema name match the original input
+      // And check special case - schema with special name in quotes
+      boolean isSchemaNameMatch = compiledSchemaPattern != null && (compiledSchemaPattern.matcher(schemaName).matches()
+          || (schemaName.startsWith("\"") && schemaName.endsWith("\"")
```

Comment:
Please put brackets after OR condition. Basically double quote condition and region stuff should be in a bracket.

Reply:
Yep, agree, mia culpa.

Reply:
![2022-09-27 00_18_18-snowflake-jdbc  F__Java Projects_snowflake_jdbc_snowflake-jdbc  -  _src_main_j](https://user-images.githubusercontent.com/45152336/192382495-524a3569-5791-41c6-9fa9-f5e85212a7f4.png)

But it is in brackets, as I can see.

Reply:
How about this way:
boolean isSchemaNameMatch = compiledSchemaPattern != null && (compiledSchemaPattern.matcher(schemaName).matches()
          || (schemaName.startsWith("\"") && schemaName.endsWith("\"")
               && compiledSchemaPattern.matcher(schemaName).region(1, schemaName.length() - 1).matches()
            )
 );

So more clear.

Reply:
Ok, if it will be handy.

Reply:
Looks good to me. Let's wait for checks to finish. Then we can merge.

Reply:
ChekStyle failed, so I did